apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserServiceImpl.java

```diff
@@ -453,6 +453,62 @@ public Uni<String> createOrUpdateUserByUserId(AddUserRoleDto userDto, String use
     }
 
+    public Uni<String> createUserByUserId(AddUserRoleDto userDto, String userId, LoggedUser loggedUser) {
+        var userInstitutionFilters = UserInstitutionFilter.builder().userId(userId).institutionId(userDto.getInstitutionId()).build().constructMap();
+        var productFilters = OnboardedProductFilter.builder().productId(userDto.getProduct().getProductId()).status(List.of(ACTIVE)).build().constructMap();
+        Map<String, Object> queryParameter = userUtils.retrieveMapForFilter(userInstitutionFilters, productFilters);
+        return userInstitutionService.retrieveFirstFilteredUserInstitution(queryParameter)
+                .onItem().transformToUni(userInstitution -> {
+                    if (Objects.nonNull(userInstitution)) {
+                        log.info("User with userId: {} has already onboarded for product {}. Proceeding with check role", userId, userDto.getProduct().getProductId());
+                        PartyRole roleOnProduct = retrieveUserRoleOnProduct(userInstitution, userDto.getProduct().getProductId());
+                        return evaluateRoleAndCreateOrUpdateUserByUserId(userDto, userId, loggedUser, roleOnProduct);
+                    } else {
+                        log.info("User with userId: {} has not onboarded for product {}. Proceeding with create", userId, userDto.getProduct().getProductId());
+                        return createOrUpdateUserByUserId(userDto, userId, loggedUser);
+                    }
+                })
+                .onFailure().invoke(exception -> log.error("Error during createOrUpdateManagerByUserId for userId: {}, institutionId: {}: {}", userId, userDto.getInstitutionId(), exception.getMessage(), exception));
+    }
+
+    /**
+     * The evaluateRoleAndCreateOrUpdateUserByUserId method is designed to evaluate the role of a user for a specific product
+     * and either create or update the user's role based on certain conditions.
+     * First, the method checks if the roleOnProduct is null. If it is, we need to create new role for the user on selected product.
+     * Next, the method compares the role specified in userDto with the existing roleOnProduct.
+     * If the new role is lower than the existing role (es. new role DELEGATE, old role MANAGER), it then calls
+     * we have to delete the old role, and subsequently we create new role for the user on selected product.
+     * If the new role is equals or higher than the existing role, throw a UserRoleAlreadyPresentException to indicate that we need to keep the old role.
+     */
+    private Uni<String> evaluateRoleAndCreateOrUpdateUserByUserId(AddUserRoleDto userDto, String userId, LoggedUser loggedUser, PartyRole roleOnProduct) {
+        if (Objects.isNull(roleOnProduct)) {
+            return createOrUpdateUserByUserId(userDto, userId, loggedUser);
+        }else if(PartyRole.valueOf(userDto.getProduct().getRole()).compareTo(roleOnProduct) < 0){
```

Comment:
`PartyRole.valueOf(userDto.getProduct().getRole())` without handling potential `IllegalArgumentException` if the role string is invalid....Before using valueOf, validate that the role string is a valid enum constant, for ex:
```

PartyRole newRole;
try {
    newRole = PartyRole.valueOf(userDto.getProduct().getRole());
} catch (IllegalArgumentException e) {
    throw new InvalidRequestException("Invalid role: " + userDto.getProduct().getRole());
}

```